features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/TimelineItemEventRow.kt

```diff
@@ -103,6 +104,9 @@ fun TimelineItemEventRow(
     fun onReactionClicked(emoji: String) =
         onReactionClick(emoji, event)
 
+    fun onMoreReactionsClicked() =
```

Comment:
This function might pass an outdated version of `event` to `onMoreReactionsClick`, but I guess it's not really and issue since we're just using the event id. Still, using a lambda here would make this less error-prone (see https://github.com/vector-im/element-x-android/pull/771#discussion_r1252736583).

Reply:
Good point, I actually ran into this exact issue just now while working on something else. 

I think the local function captures the outer function parameter the first time it's called but doesn't get updated on recomposition.

Maybe we should ensure local functions do not reference arguments to the parent composable to avoid subtle bugs like this?